message StrategyPacket {  
  enum Play {
    /// List all plays here
     Stop=0;
    PositionGather=1;
    Offense1  =2;
    Offense2  =3;
    Defense1  =4;
    Defense2  =5;
    PositionOurKickoff  =6;
    PositionOppKickoff  =7;
    PositionOurFreeKick  =8;
    PositionOppFreeKick  =9;
    PositionOurFreeBall  =10;
    PositionOppFreeBall  =11;
    PositionOurPenalty  =12;
    PositionOppPenalty  =13;
    PositionOurGoalKick  =14;
    PositionOppGoalKick  =15;
    Kickoff  =16;
    OurFreeKick  =17;
    OppFreeKick  =18;
    OurFreeBall  =19;
    OppFreeBall  =20;
    PenaltyOur  =21;
    PenaltyOpp  =22;
    OurGoalKick  =23;
    OppGoalKick  =24;
    TakeGoalKick  =25;
    TestPlay  =26;
    SetPosition  =27;
    SuperOffense  =28;
    SuperDefense  =29;
    None  =30;
  }
  optional Play play = 1;

  enum TacticID {
    Block = 0;            // Defend the goal a particular distance from the goal
    ChargeBall =1;           // Set-play tactic
    CoverGoal =2;
    DragToGoal =3;           //Go to the ball and start dragging the ballto goal
    DefendLine =4;           // Defend the ball from crossing a particular line on the field
    DefendPoint =5;          // Defend the ball from a particular point on the field
    GoalieOur =6;            // A special tactic just for the goalie for our side
    GoalieOpp =7;            // Grant Goalie to opponent team. only for testing purpose.
    MarkBot =8;              // Mark an opponent preventing them from getting the ball; getting to the goal; blocking a shot
    Pass = 9;                 // Pass to a point
    //Kickoff = 10;
    Position =11;             // Go to the given position
    PositionForPenalty =12;   // Set-play tactic
   // PositionForReceive;   // Pass Receive check
    PositionForStart =13;     // Set-play tactic
    ReceiveBall = 14;          // Receive Pass
    Defend =15;              // Defend the ball from coming to our side
    Attack =16;               // Attck the ball towards the goal      
    Steal =17;                // Manipulate the ball to remove possession of it from another robot
    //Stop =18;                 // Stop the bot
    Velocity =19;             // Move at a fixed velocity
  }
  message Tactic {
    required TacticID tID =1;
    required int32 botID =2;
  }
  optional Tactic tactic = 2;

  enum Which {
    PLAY =1;
    TACTIC =2;
  }
  required Which which = 3;
}
